centreon/www/include/configuration/configObject/service/DB-Func.php

```diff
@@ -1710,6 +1717,482 @@ function updateServiceInDBForOnPrem($serviceId = null, $massiveChange = false, $
+ *
+ * @param array $submittedValues
+ *
+ * @throws \Exception
+ * @throws \LogicException
+ * @throws CentreonDbException
+ * @throws PDOException
+ * @throws TransportExceptionInterface
+ * @throws RedirectionExceptionInterface
+ * @throws ClientExceptionInterface
+ * @throws ServerExceptionInterface
+ *
+ * @return int
+ */
+function insertServiceTemplateForOnPremise(array $submittedValues = []) : int
+{
+    global $centreon, $basePath;
+
+    try {
+        $serviceId = insertServiceTemplateByApi(
+            submittedValues: $submittedValues,
+            isCloudPlatform: false,
+            basePath: $basePath
+        );
+
+        insertServiceTemplateAdditionalOptions($serviceId, $submittedValues);
+        updateServiceContactGroup($serviceId, $submittedValues);
+        updateServiceContact($serviceId, $submittedValues);
+        updateServiceNotifs($serviceId, $submittedValues);
+        updateServiceHost($serviceId, $submittedValues);
+        updateServiceServiceGroup($serviceId, $submittedValues);
+        updateServiceTrap($serviceId, $submittedValues);
+        updateServiceCategories($serviceId, $submittedValues);
+        signalConfigurationChange('service', $serviceId);
+        $centreon->user->access->updateACL(
+            [
+                'type' => 'SERVICE',
+                'id' => $serviceId,
+                'action' => 'ADD'
+            ]
+        );
+
+        return $serviceId;
+    } catch (\Exception $exception) {
+        CentreonLog::create()->error(
+            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
+            message: "Error while creating service template: {$exception->getMessage()}",
+            customContext: ['service template Id' => $serviceId ?? null, 'basePath' => $basePath],
+            exception: $exception
+        );
+
+        throw $exception;
+    }
+}
+
+/**
+ * Inserts a new service template into the database by calling the API.
+ *
+ * @param array $submittedValues
+ * @param bool $isCloudPlatform
+ * @param string $basePath
+ *
+ * @throws \Exception
+ * @throws \LogicException
+ * @throws PDOException
+ * @throws TransportExceptionInterface
+ * @throws RedirectionExceptionInterface
+ * @throws ClientExceptionInterface
+ * @throws ServerExceptionInterface
+ *
+ * @return int
+ */
+function insertServiceTemplateByApi(
+    array $submittedValues = [],
+    bool $isCloudPlatform = false,
+    string $basePath
+): int {
+    $kernel = Kernel::createForWeb();
+    $router = $kernel->getContainer()->get(Router::class) ?? throw new \LogicException('Router not found');
+
+    if ($isCloudPlatform) {
+        $payload = getServiceTemplatePayload($submittedValues, true);
+    } else {
+        $payload = getServiceTemplatePayload($submittedValues);
+    }
+
+    $url = $router->generate(
+        'AddServiceTemplate',
+        $basePath ? ['base_uri' => $basePath] : [],
+        UrlGeneratorInterface::ABSOLUTE_URL
+    );
+
+    return callApi($url, 'POST', $payload);
```

Comment:
As you perform a POST you should check that the return of callApi is not null, and throw an exception if it is.